packages/hub/routes/merchant-infos.ts

```diff
@@ -90,10 +95,22 @@ export default class MerchantInfosRoute {
       ownerAddress: ctx.state.userAddress,
     };
 
-    await this.merchantInfoQueries.insert(merchantInfo);
+    let db = await this.databaseManager.getClient();
+    let merchantInfoId, cardSpaceId;
+
+    await inTransaction(db, async () => {
```

Comment:
This is so that we don't end up with a merchant without a card space, in case card space insert fails.